subxt/src/tx/tx_client.rs

```diff
@@ -0,0 +1,331 @@
+            .await
+    }
+
+    /// Creates and signs an extrinsic and submits it to the chain. Passes default parameters
+    /// to construct the "signed extra" and "additional" payloads needed by the extrinsic.
+    ///
+    /// Returns a [`TxProgress`], which can be used to track the status of the transaction
+    /// and obtain details about it, once it has made it into a block.
+    pub async fn sign_and_submit_then_watch_default<Call>(
+        &self,
+        call: &Call,
+        signer: &(dyn Signer<T> + Send + Sync),
+    ) -> Result<TxProgress<T, C>, Error>
+    where
+        Call: TxPayload,
+        <T::ExtrinsicParams as ExtrinsicParams<T::Index, T::Hash>>::OtherParams: Default,
+    {
+        self.sign_and_submit_then_watch(call, signer, Default::default())
+            .await
+    }
+
+    /// Creates and signs an extrinsic and submits it to the chain.
+    ///
+    /// Returns a [`TxProgress`], which can be used to track the status of the transaction
+    /// and obtain details about it, once it has made it into a block.
+    pub async fn sign_and_submit_then_watch<Call>(
+        &self,
+        call: &Call,
+        signer: &(dyn Signer<T> + Send + Sync),
+        other_params: <T::ExtrinsicParams as ExtrinsicParams<T::Index, T::Hash>>::OtherParams,
+    ) -> Result<TxProgress<T, C>, Error>
+    where
+        Call: TxPayload,
+    {
+        self.create_signed(call, signer, other_params)
+            .await?
+            .submit_and_watch()
+            .await
+    }
+
+    /// Creates and signs an extrinsic and submits to the chain for block inclusion. Passes
+    /// default parameters to construct the "signed extra" and "additional" payloads needed
+    /// by the extrinsic.
+    ///
+    /// Returns `Ok` with the extrinsic hash if it is valid extrinsic.
+    ///
+    /// # Note
+    ///
+    /// Success does not mean the extrinsic has been included in the block, just that it is valid
+    /// and has been included in the transaction pool.
+    pub async fn sign_and_submit_default<Call>(
+        &self,
+        call: &Call,
+        signer: &(dyn Signer<T> + Send + Sync),
+    ) -> Result<T::Hash, Error>
+    where
+        Call: TxPayload,
+        <T::ExtrinsicParams as ExtrinsicParams<T::Index, T::Hash>>::OtherParams: Default,
+    {
+        self.sign_and_submit(call, signer, Default::default()).await
+    }
+
+    /// Creates and signs an extrinsic and submits to the chain for block inclusion.
+    ///
+    /// Returns `Ok` with the extrinsic hash if it is valid extrinsic.
+    ///
+    /// # Note
+    ///
+    /// Success does not mean the extrinsic has been included in the block, just that it is valid
+    /// and has been included in the transaction pool.
+    pub async fn sign_and_submit<Call>(
+        &self,
+        call: &Call,
+        signer: &(dyn Signer<T> + Send + Sync),
+        other_params: <T::ExtrinsicParams as ExtrinsicParams<T::Index, T::Hash>>::OtherParams,
```

Comment:
I had a go at this, and where I landed was:

- If we can accept a reference to OtherParams when building ExtrinsicParams that'd be nice, and probably reflexct how it's used the best. But it turns out to be a pain to do so the complexity didn't feel worth while.
- If we don't, then we need to do a `.clone()` call internally, and I'd rather push that to the user (and then impls can be copy potentially and it's no biggie, or they aren't and better the user has the chance to pass ownership rather than force a clone.

So for now I'd suggest elaving it as it is, but we might be able to revisit and improve!